src/migration/1704948666033-GhostUserProtect.ts

```diff
@@ -0,0 +1,15 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class GhostUserProtect1704948666033 implements MigrationInterface {
+    name = 'GhostUserProtect1704948666033'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+      await queryRunner.query(`CREATE RULE prototect_ghostuser_deletion AS ON DELETE TO "user" WHERE old.id IN ('404') DO INSTEAD nothing;`);
```

Comment:
Can replaced `IN ('404')` with `= '404'`. Just wanted to check if there were any other special users we want to protect.

Reply:
It's ok to use IN for this purpose

Reply:
Think so too. Also makes it easier (like 2 seconds saved ⏲️) in the future if we want to expand this rule